core/src/scene/scene.h

```diff
@@ -25,183 +30,257 @@ namespace Tangram {
 class DataLayer;
 class FeatureSelection;
 class FontContext;
+class FrameBuffer;
+class Importer;
+class LabelManager;
 class Light;
 class MapProjection;
+class MarkerManager;
 class Platform;
 class SceneLayer;
+class SelectionQuery;
 class Style;
 class Texture;
 class TileSource;
-class ZipArchive;
+struct SceneLoader;
 
-// Delimiter used in sceneloader for style params and layer-sublayer naming
-const std::string DELIMITER = ":";
+struct SceneCamera : public Camera {
+    glm::dvec3 startPosition;
+};
 
-/* Singleton container of <Style> information
- *
- * Scene is a singleton containing the styles, lighting, and interactions defining a map scene
- */
+struct SceneFunctions : public std::vector<std::string> {
```

Comment:
OHH did not see this! .. Its a nono to inherit from std::vector, these do not have virtual destructors.

Reply:
>4. Inherit from the class and make sure language purist never see your code.
damn this didn't work..
ok there are three other options. https://www.researchgate.net/post/I_really_want_to_understand_if_there_is_any_reason_why_one_should_not_inherit_from_the_stl_vector_class_in_c2

Reply:
Haha I know it's technically OK since we're not using it polymorphically, but the inheritance adds ways to make bugs with this class and there's really no need for it.

Reply:
Its a failsafe thing.

Reply:
@tallytalwar good point about the virtual destructor - I didn't thought about this directly, but considered that this use as 'typedef + helper-functions' should be safe.